### PR TITLE
Focus state inconsistencies

### PR DESCRIPTION
**Pull request by chris-lawton**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1489171312)

### Description of Changes Made

- Add custom focus state to the logo links in the footer.
- Make all links that have an arrow as a child element transform on focus. Do not apply this transform if the user has set a preference for reduced motion.
- Remove `listing-avatar__arrow` class on some arrow icons - there were no styles for this in the codebase.
- Refactor `listing__arrow` styles to `icon--listing-arrow` so that the BEM syntax is valid.

### How to Test

- Tab through the components on the site that have an arrow in the link - CTA's, items on listing pages. 
- Tab through the components page in the styleguide

### Screenshots

<details>
  <summary>Expand to see more</summary>

Screenshot 2024-07-22 at 11 18 42
Screenshot 2024-07-22 at 11 18 46
Screenshot 2024-07-22 at 11 18 51
Screenshot 2024-07-22 at 11 19 01
Screenshot 2024-07-22 at 11 19 11

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (See https://intranet.torchbox.com/torchbox-com-project-docs for the private link, for Torchbox employees only)
- [ ] Updated field spec (See https://intranet.torchbox.com/torchbox-com-project-docs for the private link, for Torchbox employees only)
- [ ] Not required

#### Browser testing

- [x] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [ ] Not required

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [x] Automated WCAG 2.1 tests pass
- [ ] HTML validation passes
- [x] Manual WCAG 2.1 tests completed
- [ ] I have tested in a screen reader
- [x] I have tested in high-contrast mode
- [x] Any animations removed for prefers-reduced-motion
- [ ] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [x] Not required

#### Pattern library

- [ ] The pattern library component for this template displays correctly, and does not break parent templates
- [ ] The styleguide is updated if relevant
- [x] Changes are not relevant the pattern library

### CODE REVIEW

File: tbx/static_src/sass/components/_icon.scss [hunk 1]
```diff
@@ -24,6 +24,11 @@
         width: 100%;
     }
 
+    &--listing-arrow {
+        @include arrow-link-icon-styles();
+        margin-left: $spacer-mini-plus;
```
Comment (chris-lawton) [inline tbx/static_src/sass/components/_icon.scss:29]: Without this the arrow on titles and arrows that are only on one line overlap the focus ring:

<details><summary>Details</summary>
<p>


image

</p>
</details>

Comment (helenb) [reply tbx/static_src/sass/components/_icon.scss:29]: nitpick: maybe add an explanatory comment for why we need the margin.

File: tbx/static_src/sass/components/_icon.scss [hunk 2]
```diff
@@ -24,6 +24,11 @@
         width: 100%;
     }
 
+    &--listing-arrow {
```
Comment (helenb) [inline tbx/static_src/sass/components/_icon.scss:27]: Could you add `icon--listing-arrow` to the list of utility classes designed for re-use in the docs? https://github.com/torchbox/torchbox.com/blob/main/docs/front-end/utility-classes.md, third paragraph.

File: tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/event_block.html [hunk 3]
```diff
@@ -5,7 +5,7 @@
         <a class="event-block__title" href="{{ value.get_button_link }}" aria-label="{{ value.title }}">
             <h2>
                 {# The title and icon need to be on the same line with no whitespace to prevent the arrow being orphaned on a new line #}
-                <span>{{ value.title }}</span><span class="listing__tail">&nbsp;{% include "patterns/atoms/icons/icon.html" with name="arrow-wide" classname="listing-avatar__arrow listing__arrow" %}</span>
+                <span>{{ value.title }}</span><span class="listing__tail">&nbsp;{% include "patterns/atoms/icons/icon.html" with name="arrow-wide" classname="event-block__arrow icon--listing-arrow" %}</span>
```
Comment (helenb) [inline tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/event_block.html:8]: Praise: thank you for tidying up the classnames here.

**Review by helenb — APPROVED** [review]
Looks great, thanks @chris-lawton

File: tbx/static_src/sass/components/_icon.scss (hunk 1 above)
Comment (chris-lawton) [reply tbx/static_src/sass/components/_icon.scss]: Added 👍

File: tbx/static_src/sass/components/_icon.scss (hunk 2 above)
Comment (chris-lawton) [reply tbx/static_src/sass/components/_icon.scss:27]: Thanks, have updated

**Pull request merged**